apps/web/src/utils/usernames.ts

```diff
@@ -583,6 +583,28 @@ export async function getBasenameOwner(username: BaseName) {
   } catch (error) {}
 }
 
+export function buildBasenameNameExpiresContract(username: BaseName): ContractFunctionParameters {
+  const chain = getChainForBasename(username);
+  const tokenId = getTokenIdFromBasename(username);
+  return {
+    abi: BaseRegistrarAbi,
+    address: USERNAME_BASE_REGISTRAR_ADDRESSES[chain.id],
+    args: [tokenId],
+    functionName: 'nameExpires',
+  };
+}
+
+export async function getBasenameNameExpires(username: BaseName) {
+  const chain = getChainForBasename(username);
+
+  try {
+    const client = getBasenamePublicClient(chain.id);
+    const nameExpires = await client.readContract(buildBasenameNameExpiresContract(username));
+
+    return nameExpires;
+  } catch (error) {}
```

Comment:
imo utilities like this should not silently disappear their own errors.

Reply:
I think that's fair but we do that every where now, so maybe a larger task?